Userland/Libraries/LibWeb/TreeNode.h

```diff
@@ -97,6 +97,33 @@ class TreeNode : public Weakable<T> {
         return const_cast<TreeNode*>(this)->child_at_index(index);
     }
 
+    int child_index(const T& search_child)
```

Comment:
Please return `Optional<size_t>` instead of `int` with a magic `-1` value.
And let's call it `index_of_child()` to emphasize "index" a little extra.
And also, maybe we can `VERIFY(search_child.parent() == this);` while we're here :)